Card.h

```diff
@@ -0,0 +1,55 @@
+#ifndef CARD_H
+#define CARD_H
+
+class Card
+{
+public:
+    //two enums to define the card suits and ranks
+    enum Suit
+    {
+        clubs,
+        diamonds,
+        hearts,
+        spades,
+
+        max_suits,
```

Comment:
Having to add this fake enum value just so we can iterate through them seems nasty. I can't think of any good alternatives though sadly.